### PR TITLE
Fix bug in CKF and add script for KF timing test

### PR DESCRIPTION
**Pull request by XiaocongAi**
- Fix bug in CKF: before adding the track state, the last tip should be removed from the active tips. But when creating multiple states on ONE surface, the update of active tips should be done only ONCE so that states have the same previous state. In the current master branch, each call to `addSourcelinkState` will remove the last tip, which is not correct.
- Fix a bug in Pythia8 option: include the `pileup` generators as well instead of including the `hard` configurations twice.
- Clean the KF smoothing
- Add script for KF timing test (To produce plot as below)
image

### CODE REVIEW

**Comment by XiaocongAi** [comment]
@paulgessinger @asalzburger , could you help take a look and get it in?

**Comment by paulgessinger** [comment]
Can you comment what the bugs were? I'm not sure I understand just from the code.

**Comment by XiaocongAi** [comment]
@paulgessinger , sorry. The bugs are explained in the first comment.

**Comment by asalzburger** [comment]
@XiaocongAi can you rebase, please - this should remove the build error with macOS as well.

File: Core/include/Acts/Fitter/KalmanFitter.hpp [hunk 1]
```diff
@@ -1100,6 +1099,7 @@ class KalmanFitter {
     }
 
     if (!kalmanResult.result.ok()) {
+      ACTS_ERROR("KalmanFilter failed: " << kalmanResult.result.error());
       return kalmanResult.result.error();
     }
```
Comment (asalzburger) [inline Core/include/Acts/Fitter/KalmanFitter.hpp:1105]: I like the meaningful error output messages.

File: Examples/Scripts/KF_timing.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
+import ROOT
```
Comment (asalzburger) [inline Examples/Scripts/KF_timing.py:1]: I think we will have to do a cleanup of the scripts, probably as preparation for the Tutorial/Workshop.
This is unrelated to this PR though, it should go into an issue.

File: Examples/Algorithms/GeneratorsPythia8/ACTFW/Options/Pythia8Options.cpp [hunk 3]
```diff
@@ -72,7 +72,7 @@ FW::EventGenerator::Config FW::Options::readPythia8Options(
        Pythia8Generator::makeFunction(hard, lvl)},
       {PoissonMultiplicityGenerator{mu},
        GaussianVertexGenerator{{vtxStdXY, vtxStdXY, vtxStdZ, vtxStdT}},
-       Pythia8Generator::makeFunction(hard, lvl)},
+       Pythia8Generator::makeFunction(pileup, lvl)},
```
Comment (asalzburger) [inline Examples/Algorithms/GeneratorsPythia8/ACTFW/Options/Pythia8Options.cpp:75]: Well spotted!

File: Core/include/Acts/Fitter/KalmanFitter.hpp [hunk 4]
```diff
@@ -220,13 +226,6 @@ class KalmanFitter {
   /// Owned logging instance
   std::shared_ptr<const Logger> m_logger;
 
-  /// The navigator type
-  using KalmanNavigator = typename decltype(m_propagator)::Navigator;
-
-  /// The navigator has DirectNavigator type or not
-  static constexpr bool isDirectNavigator =
-      std::is_same<KalmanNavigator, DirectNavigator>::value;
-
```
Comment (asalzburger) [inline Core/include/Acts/Fitter/KalmanFitter.hpp:229]: Just shifting lines, hence ok.

File: Core/include/Acts/Fitter/KalmanFitter.hpp [hunk 5]
```diff
@@ -802,34 +802,30 @@ class KalmanFitter {
...
       result.fittedStates.applyBackwards(result.trackTip, [&](auto st) {
-        // Smoothing will start from the last measurement state
-        if (measurementIndices.empty()) {
-          // No smoothed parameter for the last few non-measurment states
+        bool isMeasurement =
+            st.typeFlags().test(TrackStateFlag::MeasurementFlag);
+        if (isMeasurement) {
+          measurementIndices.emplace_back(st.index());
+        } else if (measurementIndices.empty()) {
+          // No smoothed parameter if the last measurment state has not been
```
Comment (asalzburger) [inline Core/include/Acts/Fitter/KalmanFitter.hpp:816]: "No smoothed parameters if the last measurement stat has not been found yet."

File: Examples/Scripts/KF_timing.py (hunk 2 above)
Comment (XiaocongAi) [reply Examples/Scripts/KF_timing.py:1]: That's a good idea!

File: Core/include/Acts/Fitter/KalmanFitter.hpp (hunk 5 above)
Comment (XiaocongAi) [reply Core/include/Acts/Fitter/KalmanFitter.hpp]: Fixed.
